### PR TITLE
Add unit tests to openshift sdn controller logs gatherer

### PR DESCRIPTION
**Pull request by ncaak**
This PR implements a new data enhancement to...

## Categories

- [ ] Bugfix
- [ ] Data Enhancement
- [ ] Feature
- [ ] Backporting
- [X] Others (CI, Infrastructure, Documentation)

## Sample Archive

- N/A

## Documentation

- N/A

## Unit Tests

- `pkg/gatherers/clusterconfig/gather_openshift_sdn_controller_logs_test.go`

## Privacy

Yes. There are no sensitive data in the newly collected information.

## Changelog

Yes

####  Renamed file
* openshift_sdn_controller_logs.go -> gather_openshift_sdn_controller_logs.go

## References

[CCXDEV-10124](https://issues.redhat.com/browse/CCXDEV-10124)

### CODE REVIEW

File: pkg/gatherers/clusterconfig/gather_openshift_sdn_controller_logs_test.go [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+import (
+	"bufio"
+	"strings"
+	"testing"
+
+	"github.com/openshift/insights-operator/pkg/gatherers/common"
+	"github.com/stretchr/testify/assert"
+)
+
+func Test_GatherOpenshiftSDNControllerLogs(t *testing.T) {
```
[the remark this answers is not in the document]
Comment (ncaak) [reply pkg/gatherers/clusterconfig/gather_openshift_sdn_controller_logs_test.go:12]: Do you mean like in [here](https://github.com/openshift/insights-operator/blob/master/pkg/gatherers/clusterconfig/operators_pods_and_events_test.go#L43)?

[the remark this answers is not in the document]
Comment (rluders) [reply pkg/gatherers/clusterconfig/gather_openshift_sdn_controller_logs_test.go:12]: Yes! :)

File: pkg/gatherers/clusterconfig/gather_openshift_sdn_controller_logs_test.go [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+		},
+		{
+			name:     "'Node is back online' search matches successfully",
+			logline:  mockLogline("Node 'test' is back online"),
+			expected: "Node 'test' is back online",
+		},
+	}
+
+	for _, unitTest := range tests {
+		t.Run(unitTest.name, func(t *testing.T) {
```
Comment (rluders) [inline pkg/gatherers/clusterconfig/gather_openshift_sdn_controller_logs_test.go:52]: If could probably run these tests in Parallel as well, but it is up to you:

```go
tt := unitTest
t.Run(tt.name, func(t *testing.T) {
    t.Parallel()
```

Comment (ncaak) [reply pkg/gatherers/clusterconfig/gather_openshift_sdn_controller_logs_test.go:52]: Never did, I'll take a look and update this PR, thanks.

**Comment by ncaak** [comment]
/unhold

Fixed tests a bit to work in parallel. Apparently helper function was messing with the tests if throw in parallel.
Including minor refactor.

**Review by rluders — APPROVED** [review]
/lgtm

**Pull request merged**